core/config/v2/types.go

```diff
@@ -1117,3 +1279,24 @@ func (m *MercurySecrets) ValidateConfig() (err error) {
 type ThresholdKeyShareSecrets struct {
 	ThresholdKeyShare *models.Secret
 }
+
+func (t *ThresholdKeyShareSecrets) SetFrom(f *ThresholdKeyShareSecrets) (err error) {
+	err = t.validateMerge(f)
+	if err != nil {
+		return err
+	}
+
+	if v := f.ThresholdKeyShare; v != nil {
+		t.ThresholdKeyShare = v
+	}
+
+	return nil
+}
+
+func (t *ThresholdKeyShareSecrets) validateMerge(f *ThresholdKeyShareSecrets) (err error) {
+	if t.ThresholdKeyShare != nil && f.ThresholdKeyShare != nil {
+		err = multierr.Append(err, ErrOverride{Name: "ThresholdKeyShare", Msg: "overrides are not allowed for multiple secrets files"})
```

Comment:
For the users, would it make more sense to describe these as 'duplicates'?
```suggestion
		err = multierr.Append(err, ErrOverride{Name: "ThresholdKeyShare", Msg: "duplicates are not allowed with multiple secrets files"})
```

Reply:
Hmm, good question. IMO I don't think that would be as precise - the check here is not for whether there is duplicate field values. The check is for whether the same field is specified across two separate files. The field value could or could not be the same.

Reply:
Ah yeah we usually mean duplicate values. Would clarifying 'duplicate fields' help? I'm worried that 'override' is not intuitive.